rpcs3/rpcs3qt/trophy_notification_frame.cpp

```diff
@@ -0,0 +1,65 @@
+#include "trophy_notification_frame.h"
+
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QTimer>
+
+static const int TROPHY_TIMEOUT_MS = 2500;
+
+inline QString qstr(const std::string& _in) { return QString::fromUtf8(_in.data(), _in.size()); }
+
+trophy_notification_frame::trophy_notification_frame(const std::vector<uchar>& imgBuffer, const SceNpTrophyDetails& trophy, int height) : QWidget()
+{
+	setWindowFlags(Qt::FramelessWindowHint);
+
+	// Fill the background with black
+	QPalette black_background;
+	black_background.setColor(QPalette::Window, Qt::black);
+	black_background.setColor(QPalette::WindowText, Qt::white);
+
+	// Make the label
+	QLabel* trophyImgLabel = new QLabel;
+	trophyImgLabel->setAutoFillBackground(true);
+	trophyImgLabel->setPalette(black_background);
+
+	QImage trophyImg;
+	if (imgBuffer.size() > 0 && trophyImg.loadFromData((uchar*)&imgBuffer[0], imgBuffer.size(), "PNG"))
+	{
+		trophyImg = trophyImg.scaledToHeight(height); // I might consider adding ability to change size since on hidpi this will be rather small.
+		trophyImgLabel->setPixmap(QPixmap::fromImage(trophyImg));
+	}
+	else
+	{
+		// This looks hideous, but it's a good placeholder.
+		trophyImgLabel->setPixmap(QPixmap::fromImage(QImage(":/rpcs3.ico")));
+	}
+
+	QLabel* trophyName = new QLabel;
+	trophyName->setWordWrap(true);
+	trophyName->setAlignment(Qt::AlignCenter);
+
+	QString trophyType = "";
+	switch (trophy.trophyGrade)
```

Comment:
I get an exception here because the variable "trophy" contains only garbage at this point (it becomes mangled randomly between the start of the method and this point). The issue does not occur if the variable is passed by value instead of by reference so it should be related to how the memory for the variable is initially allocated.

Reply:
yes, i have same thinkl flash already informed ^^

Reply:
Yeah thanks for letting me know, guys.  It's an issue of how the lambda is scoped.  Just pushed a commit fixing it if you're curious what happened.